addons/crm_survey/tests/test_crm_survey.py

```diff
@@ -0,0 +1,257 @@
+
+        ### Check if the last created lead was from the survey
+        last_lead_created = self.env['crm.lead'].search([], order='create_date desc', limit=1)
+        self.assertNotEqual(last_lead_created.name, f"{user.display_name}'s survey results")
+
+    def test_not_connected_account_access_with_lead_generation_answer(self):
+        # Step 3 : Public access + lead generation
+        survey = self._create_lead_qualification_survey(is_in_sales_team=True)
+
+        # Start page
+        self._access_start(survey)
+        user_inputs = self.env['survey.user_input'].search([('survey_id', '=', survey.id)])
+        answer_token = user_inputs.access_token
+
+        # First page
+        response = self._access_page(survey, answer_token)
+        csrf_token = self._find_csrf_token(response.text)
+        self._access_begin(survey, answer_token)
+
+        # Answers
+        with self.with_user('survey_user'):
+            question_ids = list(survey.question_ids)
+            self._answer_question(question_ids[0], question_ids[0].suggested_answer_ids.ids[0], answer_token, csrf_token)
+            self._answer_question(question_ids[1], question_ids[1].suggested_answer_ids.ids[0], answer_token, csrf_token)
+            self._answer_question(question_ids[2], question_ids[2].suggested_answer_ids.ids[0], answer_token, csrf_token)
+            self._answer_question(question_ids[3], question_ids[3].suggested_answer_ids.ids[0], answer_token, csrf_token)
+            self._answer_question(question_ids[4], "harry@potter.poudlard", answer_token, csrf_token)
+
+        ### Check if the last created lead was from the survey
+        last_lead_created = self.env['crm.lead'].search([], order='create_date desc', limit=1)
+        self.assertEqual(last_lead_created.name, "harry@potter.poudlard's survey results")
+
+        # Ensure that the result values are present in lead description
+        description = last_lead_created.description
+        for answer in user_inputs.user_input_line_ids:
+            self.assertIn(answer._get_answer_value(), description)
+
+        # Ensure contact, salesperson, medium, source, email and contact name are rights
+        self.assertFalse(last_lead_created.partner_id.id)  # Public user
+        self.assertEqual(last_lead_created.user_id.id, survey.user_id.id)  # Survey created by a sales team person
+        self.assertEqual(last_lead_created.medium_id.name, "Survey")
+        self.assertEqual(last_lead_created.source_id.name, survey.title)
+        self.assertEqual(last_lead_created.email_from, "harry@potter.poudlard")
+        self.assertEqual(False, last_lead_created.contact_name)
+
+        # Ensure Odoobot created the lead
+        for message in last_lead_created.message_ids:
+            self.assertEqual(message.author_id.name, self.env.ref('base.user_root').name)
+
+    # Override common test survey class for adding "is_create_lead" attribute
+    def _add_question(self, page, name, qtype, **kwargs):
```

Comment:
No super call? no good ^^ this can lead to some issues, and a lot of duplication as you probably notice here. As odoo is extremely modular, we want the method to be able to work if we do more overrides. So we always want to call super(), to make sure that all behaviors and overrides are effective. Rule of thumb :)

Instead, we should probably call super, and modify the result. Or even, do not modify the method at all and modify the answers directly in the records, once created using _add_question in your other method :)